### PR TITLE
Pull pinning migrations directly from GH central source

### PR DESCRIPTION
**Pull request by CJ-Wright**
Checklist
* [ ] Added a ``news`` entry

### CODE REVIEW

File: conda_smithy/configure_feedstock.py [hunk 1]
```diff
@@ -1630,6 +1634,27 @@ def get_migrations_in_dir(migrations_root):
...
+    with tempfile.TemporaryDirectory() as td:
+        tdn = td.name
+        # TODO: potentially pull git args from a config (in case you wanted
+        #  to point at a different remote/branch/etc.)
+        subprocess.call(['git', 'clone', '--depth', '1', 'https://github.com/conda-forge/conda-forge-pinning-feedstock.git', tdn])
+        cfp_migration_dir = os.path.join(tdn, 'recipe', 'migrations')
+        if not os.path.exists(cfp_migration_dir):
+            cfp_migration_dir = None
+        migrations = get_migrations_in_dir(cfp_migration_dir)
+    return migrations
```
Comment (isuruf) [inline conda_smithy/configure_feedstock.py]: You are returning the file paths from a temporary directory which is going to get deleted after the context manager exists

Comment (CJ-Wright) [reply conda_smithy/configure_feedstock.py]: Ah, sorry I thought `get_migrations_in_dir` returned the actual migrations

Comment (CJ-Wright) [reply conda_smithy/configure_feedstock.py]: Now the paths should be valid until they are parsed at which time we no longer need them.

**Comment by beckermr** [comment]
closing as we solved these issues

**Pull request closed without merging**